### PR TITLE
Icon-only button sizing and A11Y - RSC-398

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-398

### CODE REVIEW

File: gallery/src/components/NxButton/NxButtonPage.tsx [hunk 1]
```diff
@@ -51,6 +51,18 @@ export default function NxButtonPage() {
...
               <NxTableCell>The variant of button. See examples of each variant below.</NxTableCell>
             </NxTableRow>
+            <NxTableRow>
+              <NxTableCell>title</NxTableCell>
+              <NxTableCell>string</NxTableCell>
+              <NxTableCell>Required on icon-only buttons</NxTableCell>
+              <NxTableCell>Empty</NxTableCell>
+              <NxTableCell>
+                A string to render as a tooltip and accessibility label for the button. This is generally not necessary
+                for buttons that include text content, but icon-only buttons should use this to make the buttons meaning
```
Comment (glenhunter) [inline gallery/src/components/NxButton/NxButtonPage.tsx:61]:
```suggestion
                for buttons that include text content, but icon-only buttons should use this to make the button's meaning
```

File: gallery/src/components/NxButton/NxButtonPage.tsx [hunk 2]
```diff
@@ -96,8 +108,8 @@ export default function NxButtonPage() {
                           id="nx-button-icon-only-example"
                           liveExample={NxButtonIconOnlyExample}
                           codeExamples={nxButtonIconOnlyCode}>
-        An example of a button containing only an icon. For accessibility purposes it is important to add an
-        aria-label for the screen reader to interpret the content correctly.
+        An example of a button containing only an icon. For accessibility purposes it is important to use the
+        title prop for the screen reader to interpret the content correctly.
```
Comment (glenhunter) [inline gallery/src/components/NxButton/NxButtonPage.tsx:112]:
```suggestion
        title prop for a screen reader to interpret the content correctly.
```

File: lib/src/base-styles/_nx-btn.scss [hunk 3]
```diff
@@ -20,6 +20,7 @@
   @include container-horizontal;
   @extend %nx-btn--disabled;
 
+  box-sizing: content-box;
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-btn.scss:23]: Why this change?

**Comment by rpokorny** [comment]
Remaining question for @dturner32 : which size should icon-only buttons in the page header be? There were several applitools tests of `NxNexusPageHeader` that changed in this PR

File: lib/src/base-styles/_nx-btn.scss (hunk 3 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-btn.scss]: > Why this change?

So for one thing, it turns out the UA stylesheet (in Chrome at least) had this set to `border-box` by default for `<button>`s.  I think I changed it to make the the width calculations here and in the mixin slightly simpler. However it also made the min-width calculation in the pagination button more complicated, and I just realized I have that calculation wrong (forgot borders).  So I've now changed this including removing this `box-sizing`. 3568b477

**Comment by glenhunter** [comment]
Safari is unhappy with the buttons in the dropdowns. I checked Master and it's fine.

Screen Shot 2021-04-16 at 12 22 32 PM

**Comment by glenhunter** [comment]
Safari also disrespects the 42px icon only button:

Screen Shot 2021-04-16 at 2 21 58 PM

**Comment by rpokorny** [comment]
> Safari is unhappy...
> Safari also disrespects...

Safari placated here: 9549551d

**Review by glenhunter — APPROVED** [review]
